src/tools/tidy/src/main.rs

```diff
@@ -7,19 +7,49 @@
 use std::collections::VecDeque;
 use std::num::NonZeroUsize;
 use std::path::PathBuf;
+use std::process::Command;
 use std::str::FromStr;
 use std::sync::atomic::{AtomicBool, Ordering};
 use std::thread::{self, ScopedJoinHandle, scope};
 use std::{env, process};
 
 use tidy::*;
 
+// We set the `BASE_COMMIT` environment variable
+fn set_env_base_commit() {
+    if let Ok(base_commit) = env::var("BASE_COMMIT") {
+        // Nothing to be done!
+        println!("base commit is {base_commit}");
+        return;
+    }
+    // If first commit is from `bors@rust-lang.org`, then we need to skip it as it means we're in
+    // the CI and it's the merge commit generated to test the PR.
+    let output = Command::new("git")
```

Comment:
This logic is much more complicated than it might seem. Could you instead reuse https://github.com/rust-lang/rust/blob/a4b9a1b4dabe3bbeb3f8db19fbfa02302ca73501/src/build_helper/src/git.rs#L201? This is more complicated than it might seem :sweat_smile: